__Задача 1__

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Событие А

А=вероятность, что из 4-х выбранных карт хотя бы один туз 

Событие $\overline{A}$ -из 4-х карт ни одного туза

Находим общее число исходов выбрать 4 карты из 52

In [1]:
from math import factorial
def sochet(n,k):
    return int(factorial(n)/(factorial(k)*factorial(n-k)))

In [2]:
N=sochet(52,4)

Находим число исходов при которых выбрано 4 карты не тузы 

In [3]:
N_not_ace=sochet((52-4),4)

веротяность того, что из 52-х карт выбрано 4 и среди них нет ни одного туза

In [4]:
P_not_ace=N_not_ace/N
P_not_ace

0.7187367254594146

Вероятность что выпадет хтя бы один туз
$$ P(A)=1-P(\overline{A})$$


In [5]:
P=1-P_not_ace
P

0.2812632745405854

In [13]:
sochet(52,4)-sochet(48,4)

76145

###### Ответ: число исходов, соответствующих тому, что был вытянут хотя бы один туз 0.2812632745405854

__Задача 2__

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?

###### Первая часть задачи. Два определенных человека сядут рядом (A).

1.1.первый человек сел с краю скамейки - 2 исхода из 7 мест

1.2 второй человек сел рядом с первым - 1 исход из 6 оставшихся мест

2.1 первый человек сел в любом месте кроме крайних двух - 5 исходов из 7 мест

2.2 второй сел рядом с первым - 2 исхода из 6 оставшихся мест

Общая вероятность:
$$ P(A)=\dfrac{2}{7}\cdot \dfrac{1}{6}+ \dfrac{5}{7}\cdot \dfrac{2}{6}=\dfrac{12}{42}=\dfrac{2}{7}$$

In [31]:
(2/7)*(1/6)+(5/7)*(2/6)

0.2857142857142857

In [14]:
2/7

0.2857142857142857

###### Вторая часть задачи. три определённых человека (B)?
Решение через комбинаторику.

Количество перестановок 3-х человек $$  n(3)=3!=6 $$

Количество случаев, когда 3 человека сидят рядом на скамейке на 7 мест: $$ 7-2=5 $$

Количество благоприятных исходов: $$5\cdot 3!=30$$

Общее количество размещений из 3 человек по 7 местам: $$ \dfrac{7!}{(7-3)!}=5\cdot6\cdot7=210$$

Вероятность P(B) $$P(B)=\dfrac{30}{210}=\dfrac{1}{7}$$ 

In [15]:
1/7

0.14285714285714285

__Задача 3__

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

а) решение через теорию вероятностей

In [36]:
(50*49*48)/(60*59*58)

0.5727644652250146

a) решение через комбинаторику

In [40]:
с_summary=sochet(60,3)
c_positive=sochet(50,3)
C=c_positive/с_summary
C

0.5727644652250146

б) решение через теорию вероятностей

In [43]:
(3*50*10*49)/(60*59*58)

0.35797779076563413

б) решение через комбинаторику

In [48]:
с_summary=sochet(60,3)
c_positive=sochet(50,2)
c_negative=sochet(10,1)
C=(c_positive*c_negative)/с_summary
C

0.35797779076563413

__Задача 4__

Бросается игральная кость. Пусть событие `A` - появление чётного числа, событие `B` - появление числа больше трёх. Являются ли эти события независимыми?


###### Первоначальный ответ : 
 так как четное число может быть и больше и меньше трех (пример: 2<3; 4,6>3), и обратно, число больше трех может быть как четным (4,6) так и нечетным (5). Так как требований, чтобы исполнилось сразу два события нет, то эти события независимы
 
 ###### Ответ после разбора ДЗ на 2-м уроке:
 События зависимы, так как вероятности для события А отличаются при разных исходах события В 

__Задача 5__

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

_Вопрос_: какова вероятность, что вы действительно больны ей?

_Подсказка_: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

##### Решение:

событие $B$ - я болен 

событие $\overline{B}$ - я не болен

событие $А$ - положительный результат теста

Вероятнотсь того, что результат теста положителен: $P(A)=P(A|B)\cdot P(B) + P(A|\overline{B})\cdot P(\overline{B}) $

$$P(A)=1\cdot 0,001 + 0,01\cdot 0,999=0,01099$$

In [53]:
1*0.001+0.01*0.999

0.01099

выразим через формулу Байеса
$$P(B|A) = \frac{P(A|B) \cdot P(B)}{P(A)}$$

In [54]:
0.001/0.01099

0.09099181073703368